### PR TITLE
Validate remote LDAP provider credentials

### PR DESCRIPTION
**Pull request by DavidePrincipi**
- Hide `ldapservice` account from Users & Groups page, by ignoring it in NSS db
- Display `ldapservice` credentials in local AD provider page
- Display TODO if credentials for LDAP applications are missing
- Validate LDAP settings and credentials against remote provider
- Fix StartTls on fetchdump command
- Deprecation of `nethserver-sssd-notifyclients` action: a log warning is issued

NethServer/dev#5396

### CODE REVIEW

File: root/etc/e-smith/templates/etc/sssd/sssd.conf/00template_vars [hunk 1]
```diff
@@ -8,5 +8,6 @@
 
     our $suffix = $sssd_object->baseDN();
     our $provider = $sssd{'Provider'} || 'ldap';
+    our @filter_users = qw(ldapservice);
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/sssd/sssd.conf/00template_vars]: Remove this fragment.

File: root/etc/e-smith/templates/etc/sssd/sssd.conf/30nss [hunk 2]
```diff
@@ -1 +1,7 @@
 [nss]
+{
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/sssd/sssd.conf/30nss]: Use static code:
```
filter_users = ldpaservice
```

File: root/etc/e-smith/validators/actions/ldaptestbind [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash -x
```
Comment (gsanchietti) [inline root/etc/e-smith/validators/actions/ldaptestbind]: Remove `-x` option

File: root/etc/e-smith/events/actions/nethserver-sssd-notifyclients [hunk 4]
```diff
@@ -25,13 +25,18 @@
...
-    | xargs -- rpm -qf --queryformat '%{NAME}-update\n' | sort | uniq \
-    | grep -v -E 'nethserver-(dc|sssd|directory)-update'); do
+tmp_patterns=$(mktemp)
+trap 'rm -f ${tmp_patterns}' EXIT
+
+# Prepare the list of excluded events:
+find /etc/e-smith/events/nethserver-sssd-save/ -type f -o -type l | xargs -- rpm -qf --queryformat '%{NAME}-update\n' \
+    | sort | uniq >$tmp_patterns
+
+for SUBEVENT in $(/usr/libexec/nethserver/find-sssd-deps | grep -v -F -f ${tmp_patterns}); do
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-sssd-notifyclients]: Revert to previous search command.

File: root/etc/nethserver/todos.d/20ad_credentials [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+import gettext
+import json
+import subprocess
+import sys
+
+out = ''
+
+try:
+    out = json.load(subprocess.Popen(['/sbin/e-smith/config', 'printjson', 'sssd'], stdout=subprocess.PIPE).stdout)
```
Comment (gsanchietti) [inline root/etc/nethserver/todos.d/20ad_credentials]: Disable this todo if ldap credentials are not required.

Comment (DavidePrincipi) [reply root/etc/nethserver/todos.d/20ad_credentials]: Fixed by commit "Refactor ldap-credentials-optional UI helper"

**Pull request merged**